Framework/Source/Utils/Platform/Linux/Linux.cpp

```diff
@@ -0,0 +1,477 @@
+        {
+            return std::string(cwd);
+        }
+
+        return std::string();
+    }
+
+    const std::string& getExecutableName()
+    {
+        static std::string filename;
+        filename = fs::path(program_invocation_name).filename();
+        return filename;
+    }
+
+    bool getEnvironmentVariable(const std::string& varName, std::string& value)
+    {
+        const char* val = ::getenv(varName.c_str());
+        if (val == 0)
+        {
+            return false;
+        }
+        static std::string strvar(val);
+        value = strvar;
+        return true;
+    }
+
+    std::vector<std::string> gDataDirectories =
+    {
+        // Ordering matters here, we want that while developing, resources will be loaded from the development media directory
+        std::string(getWorkingDirectory()),
+        std::string(getWorkingDirectory() + "/Data"),
+        std::string(getExecutableDirectory()),
+        std::string(getExecutableDirectory() + "/Data"),
+
+        // The local solution media folder
+        std::string(getExecutableDirectory() + "/../Media"),
+    };
+
+    const std::vector<std::string>& getDataDirectoriesList()
+    {
+        return gDataDirectories;
+    }
+
+    void addDataDirectory(const std::string& dataDir)
+    {
+        //Insert unique elements
+        if (std::find(gDataDirectories.begin(), gDataDirectories.end(), dataDir) == gDataDirectories.end())
+        {
+            gDataDirectories.push_back(dataDir);
+        }
+    }
+
+    std::string canonicalizeFilename(const std::string& filename)
+    {
+        fs::path path(replaceSubstring(filename, "\\", "/"));
+        return fs::exists(path) ? fs::canonical(path).string() : "";
+    }
+
+    bool findFileInDataDirectories(const std::string& filename, std::string& fullpath)
+    {
+        static bool bInit = false;
+        if (bInit == false)
+        {
+            std::string dataDirs;
+            if (getEnvironmentVariable("FALCOR_MEDIA_FOLDERS", dataDirs))
+            {
+                auto folders = splitString(dataDirs, ";");
+                gDataDirectories.insert(gDataDirectories.end(), folders.begin(), folders.end());
+            }
+            bInit = true;
+        }
+
+        // Check if this is an absolute path
+        if (doesFileExist(filename))
+        {
+            fullpath = canonicalizeFilename(filename);
+            return true;
+        }
+
+        for (const auto& Dir : gDataDirectories)
+        {
+            fullpath = canonicalizeFilename(Dir + '/' + filename);
+            if (doesFileExist(fullpath))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    template<bool bOpen>
+    static bool fileDialogCommon(const char* pFilters, std::string& filename)
+    {
+        if (!gtk_init_check(0, nullptr))
+        {
+            should_not_get_here();
+        }
+
+        GtkWidget *pParent = gtk_window_new(GTK_WINDOW_TOPLEVEL);
```

Comment:
Coding convention says that the `*` should be attached to the type. There are many places in this file that require fixes